lib/components/health_component.cpp

```diff
@@ -0,0 +1,7 @@
+#include "components/health_component.hpp"
+
+HealthComponent::HealthComponent(double health, DeathFunction on_death) : health(health), on_death(on_death) {}
```

Comment:
I noticed we miss a lot of const in the methods' arguments, this is one of the examples where health or the function on_death could be marked as const as no changes are made in the constructor. We might want to discuss how/ when we would like to solve this. I do not think it is something we should look at in this pull request.

Reply:
Making these variables const would do absolutely nothing as they are copied.

Reply:
Allright, that was something I did not take into consideration